### PR TITLE
implement OAS postAuthenticationPlugin

### PR DESCRIPTION
**Pull request by jeffy-mathew**
https://tyktech.atlassian.net/browse/TT-7744

### CODE REVIEW

File: apidef/oas/middleware.go [hunk 1]
```diff
@@ -882,3 +891,27 @@ func (p *PrePlugin) ExtractTo(api *apidef.APIDefinition) {
...
+	}
+
+	p.Plugins = make(CustomPlugins, len(api.CustomMiddleware.PostKeyAuth))
+	p.Plugins.Fill(api.CustomMiddleware.PostKeyAuth)
+}
+
+// ExtractTo extracts PostAuthenticationPlugin into Tyk classic api definition.
+func (p *PostAuthenticationPlugin) ExtractTo(api *apidef.APIDefinition) {
+	api.CustomMiddleware.PostKeyAuth = make([]apidef.MiddlewareDefinition, len(p.Plugins))
+	p.Plugins.ExtractTo(api.CustomMiddleware.PostKeyAuth)
```
Comment (furkansenharputlu) [inline apidef/oas/middleware.go:916]: p.Plugins check?

File: apidef/migration_test.go [hunk 2]
```diff
@@ -589,7 +589,8 @@ func TestAPIDefinition_migrateCustomPluginAuth(t *testing.T) {
 func TestSetDisabledFlags(t *testing.T) {
 	apiDef := APIDefinition{
 		CustomMiddleware: MiddlewareSection{
-			Pre: make([]MiddlewareDefinition, 1),
+			Pre:         make([]MiddlewareDefinition, 1),
+			PostKeyAuth: make([]MiddlewareDefinition, 1),
```
Comment (furkansenharputlu) [inline apidef/migration_test.go:593]: When `CustomMiddleware` is empty, the `Pre` value is already `nil`. The `disabled` should not cause a problem in empty `Definition` ?

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4712/merge
Commit: 67a5286d597153db47dabbac7b365f50129cf8dc 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/4043742074)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4712/merge
Commit: 2bd25375817adc7e03ae7377b797dfad28626245 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/4043753294)

File: apidef/migration_test.go (hunk 2 above)
Comment (jeffy-mathew) [reply apidef/migration_test.go:593]: yes, but we're testing `SetDisabledFlags` when `CustomMiddleware` is empty, we're not testing that

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4712/merge
Commit: c2a26680b739e5ad5dc8f8ff337ab0e0070821f1 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/4045431886)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4712/merge
Commit: 2bd25375817adc7e03ae7377b797dfad28626245 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/4045455271)

File: apidef/oas/middleware.go [hunk 3]
```diff
@@ -830,6 +834,9 @@ type CustomPlugin struct {
 	Path string `bson:"path" json:"path"` // required.
 	// RawBodyOnly if set to true, do not fill body in request or response object.
 	RawBodyOnly bool `bson:"rawBodyOnly,omitempty" json:"rawBodyOnly,omitempty"`
+	// RequireSession if set to true passes down the session information for plugins post authentication.
+	// RequireSession is used only with JSVM custom middleware.
+	RequireSession bool `bson:"requireSession" json:"requireSession"`
```
Comment (titpetric) [inline apidef/oas/middleware.go:839]: Missing omitempty

**Review by titpetric — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4712/merge
Commit: bc5042f418b37e3a5a6f1adda0def230a9e59f91 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/4045575901)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4712/merge
Commit: dbf25eb9a5b3bcb53ddc970cb4a9aa0ae3ccdbca 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/4045756313)

**Review by furkansenharputlu — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4712/merge
Commit: 854dee90e21d69320051b48bea34012826b11ddc 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/4054496405)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4712/merge
Commit: a82ccbcdf9442cbd04a299e96dd0f088ea943b41 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/4054512541)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4712/merge
Commit: d49eec109cb71917685f47c0a2e9b5ca3a4dfacf 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/4054571015)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4712/merge
Commit: 859b61819ce07ccc27121ae62299b6b5db185e72 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/4060774623)

**Pull request merged**